scripts/linux_share_build.sh

```diff
@@ -3,33 +3,82 @@ PROC=$((`grep -c processor /proc/cpuinfo`))
 
 function install_prerequisites() {
     TZ=America/Chicago
-    ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
-    apt-get update -y
-    apt-get install -y g++ libeigen3-dev libhdf5-dev patchelf git cmake
+    $SUDO ln -snf /usr/share/zoneinfo/$TZ /etc/localtime
+    $SUDO sh -c 'echo $TZ > /etc/timezone'
+    $SUDO apt-get update -y
+    $SUDO apt-get install -y g++ libeigen3-dev patchelf git cmake curl
 }
 
-function setup_folder() {
+
+function setup() {
+    unset LD_LIBRARY_PATH
+    
+    echo "Building the Trelis plugin in ${CURRENT}\\${PLUGIN_DIR}"
     cd ${CURRENT}
     mkdir ${PLUGIN_DIR}
-    echo "Building the Trelis plugin in ${CURRENT}\\${PLUGIN_DIR}"
-
     cd ${PLUGIN_DIR}
     PLUGIN_ABS_PATH=$(pwd)
-    ln -s $SCRIPTPATH/ ./
+    ln -s ${SCRIPTPATH}/ ./
+
+    # upload the variable in GithubAction
+    if [ "$GHA" = "ON" ]; then
+        echo "PLUGIN_ABS_PATH=$PLUGIN_ABS_PATH" >> $GITHUB_ENV
+    fi
+}
+
+function setup_var() {
+    # Setup the variables
+    if [ "$1" = "2020.2" ]; then
+        TRELIS_PATH="/opt/Coreform-Cubit-2020.2"
+        TRELIS_PKG="Coreform-Cubit-2020.2-Lin64.deb"
+        TRELIS_SDK_PKG="Coreform-Cubit-2020.2-Lin64-SDK.tar.gz"
+    elif [ "$1" = "17.1.0" ]; then
+        TRELIS_PATH="/opt/Trelis-17.1"
+        TRELIS_PKG="Trelis-17.1.0-Lin64.deb"
+        TRELIS_SDK_PKG="Trelis-SDK-17.1.0-Lin64.tar.gz"
+        CMAKE_ADDITIONAL_FLAGS="-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=0"
+    else
+        echo "unknown Trelis/Cubit version, use: \"17.1\" or \"2020.2\""
+    fi
+
+    # upload the variable in GithubAction
+    if [ "$GHA" = "ON" ]; then
+        echo "TRELIS_PATH" $TRELIS_PATH
+        echo "TRELIS_PATH=$TRELIS_PATH" >> $GITHUB_ENV
+        echo "TRELIS_PKG=$TRELIS_PKG" >> $GITHUB_ENV
+        echo "TRELIS_SDK_PKG=$TRELIS_SDK_PKG" >> $GITHUB_ENV
+        echo "CMAKE_ADDITIONAL_FLAGS=$CMAKE_ADDITIONAL_FLAGS" >> $GITHUB_ENV
+    fi
+}
+
+function build_hdf5() {
+    cd ${PLUGIN_ABS_PATH}
+    mkdir -p hdf5/bld
+    cd hdf5
+    git clone https://github.com/HDFGroup/hdf5.git -b hdf5-1_12_0
+    cd bld
+    cmake ../hdf5 -DBUILD_SHARED_LIBS:BOOL=ON
+    make -j$PROC
+    $SUDO make install
 }
 
 function build_moab() {
     cd ${PLUGIN_ABS_PATH}
     mkdir -pv moab/bld
     cd moab
     git clone https://bitbucket.org/fathomteam/moab -b Version5.1.0
+    cd moab
+    # patching MOAB CMakeLists.txt to use default find(HDF5)
+    sed -i "s/HDF5_MOAB/HDF5/" CMakeLists.txt
+    cd ..
+    #end of patch
     cd bld
     cmake ../moab -DENABLE_HDF5=ON \
-            -DCMAKE_PREFIX_PATH=/usr/lib/x86_64-linux-gnu/hdf5/serial \
```

Comment:
Were you having trouble finding HDF5 in docker or GHA? I'm curious if I might be able to try a few things out locally.

Reply:
both as soon as I install my own hdf5, I can't use the `find(HDF5_MOAB)` from the MOAB source

Reply:
that would be great, but make sure you use HDF5 1.12.0, as this is the one we shipped with Cubit 2020.2